### PR TITLE
Tarball fetcher: Fix handling of cached tarballs

### PR DESCRIPTION
**Pull request by edolstra**
# Motivation

Fixes a regression introduced in 5a9e1c0d20e2332c79fb0fd7570315a5d93041f2 where downloading a cached file causes the error "Failed to open archive (Unrecognized archive format)".

# Context

# Priorities and Process

Add :+1: to [pull requests you find important](https://github.com/NixOS/nix/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc).

The Nix maintainer team uses a [GitHub project board](https://github.com/orgs/NixOS/projects/19) to [schedule and track reviews](https://github.com/NixOS/nix/tree/master/maintainers#project-board-protocol).

### CODE REVIEW

File: src/libutil/tarfile.cc [hunk 1]
```diff
@@ -81,6 +81,7 @@ TarArchive::TarArchive(Source & source, bool raw, std::optional<std::string> com
     if (!raw) {
         archive_read_support_format_tar(archive);
         archive_read_support_format_zip(archive);
+        archive_read_support_format_empty(archive);
```
Comment (roberth) [inline src/libutil/tarfile.cc:84]: Is this for uncompressed tarballs?
Could we add a test for that?

Comment (edolstra) [reply src/libutil/tarfile.cc:84]: No, it's for empty files.

Comment (roberth) [reply src/libutil/tarfile.cc:84]: This is very surprising, maybe add a few sentences?

**Review by roberth — APPROVED** [review]
Let's merge when documented.

**Pull request merged**